### PR TITLE
[LoongArch64] Avoid using native layout info to calculate register flags for small structs where possible.

### PR DESCRIPTION
**Pull request by LuckyXu-HF**
Avoid using native layout info to calculate register flags for small structs where possible for LA64.
This is mentioned in PR https://github.com/dotnet/runtime/pull/97877

### CODE REVIEW

**Comment by LuckyXu-HF** [comment]
@t-mustafin , thank you very much for your reminder!

**Review by shushanhf — APPROVED** [review]
LGTM
@jkotas 

Thanks

File: src/coreclr/vm/methodtable.cpp [hunk 1]
```diff
@@ -2804,18 +2804,12 @@ bool MethodTable::IsLoongArch64OnlyOneField(MethodTable * pMT)
 {
```
Comment (jkotas) [inline src/coreclr/vm/methodtable.cpp:2804]: Could you please move this method outside the `TARGET_LOONGARCH64` ifdef, rename it to `MethodTable::IsOnlyOneField`, and use it for both the LoongArch and RISCV (delete `IsRiscV64OnlyOneField` method that is identical).

Comment (LuckyXu-HF) [reply src/coreclr/vm/methodtable.cpp:2802]: OK，I have already updated this, thanks very much!

Comment (LuckyXu-HF) [reply src/coreclr/vm/methodtable.cpp:2802]: @tomeksowi @clamp03 @sirntar ,this also related to RISCV, could you please also take a look?

**Review by tomeksowi — APPROVED** [review]
LGTM

**Review by sirntar — APPROVED** [review]

**Review by jkotas — APPROVED** [review]

**Review by clamp03 — APPROVED** [review]
Thank you.

**Pull request merged**